File: Array/T35_Search Insert Position/GO/main.go
package main

import "fmt"

func main() {
	nums1 := []int{1, 3, 5, 6}
	target1 := 7

	nums2 := []int{1, 3, 5, 6}
	target2 := 2

	fmt.Println("case1:", searchInsert(nums1, target1))
	fmt.Println("case2", searchInsert(nums2, target2))

}

func searchInsert(nums []int, target int) int {
	for i := 0; i < len(nums); i++ {
		if target <= nums[i] {
			return i
		}
	}

	return len(nums)
}

File: Array/T35_Search Insert Position/PYTHON/main.py
from operator import le
from turtle import right
from typing import List


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # 二分查找法
        left = 0
        right = len(nums) - 1
        while left <= right:
            middle = left + (right - left) // 2
            if target > nums[middle]:
                left = middle + 1
            elif target < nums[middle]:
                right = middle - 1
            else:
                return middle
        return right+1
